crates/formatter/tests/cases/array_alignment/after.php

```diff
@@ -0,0 +1,219 @@
+    /// Weeeee!
+    /// Weee!!!!
+    /* Weeeee! */
+    ['October',   '$0.00',   '$0.00'],
+    ['November',  '$0.00',   '$0.00'],
+    ['December',  '$0.00',   '$0.00'], // This is a comment
+]); // This is a comment
+
+show_table([ // This is a comment
+    ['Hello', 11212, 112.1, true,  $bar,  PHP_VERSION],
+    ['World', 125.1, 12,    false, $quux, PHP_VERSION_ID],
+]);
+
+show_table([[[ // This is a comment
+    ['Hello', 11212, 112.1, true,  $bar,  PHP_VERSION],
+    ['World', 125.1, 12,    false, $quux, PHP_VERSION_ID],
+]]]);
+
+show_table(array( // This is a comment
+    array('Month',     'Premium', 'Revenue'), // This is a comment
+    array('January',   '$0.00',   '$0.00'),
+    array('February',  '$0.00',   '$0.00'),
+    array('March',     '$0.00',   '$0.00'),
+    array('April',     '$0.00',   '$0.00'),
+    array('May',       '$0.00',   '$0.00'),
+    array('June',      '$0.00',   '$0.00'),
+    array('July',      '$0.00',   '$0.00'),
+    array('August',    '$0.00',   '$0.00'),
+    array('September', '$0.00',   '$0.00'),
+    array('October',   '$0.00',   '$0.00'),
+    array('November',  '$0.00',   '$0.00'),
+    array('December',  '$0.00',   '$0.00'), // This is a comment
+)); // This is a comment
+
+show_table([ // This is a comment
+    array('Hello', 11212, 112.1,  true,  $bar,  PHP_VERSION),
+    array('World', 125.1, 12,     false, $quux, PHP_VERSION_ID),
+    array('!!',    125.1, 123512, false, $bar,  PHP_VERSION_ID),
+]);
+
+// This table contains a very long string so it won't be formatted as a table
+show_table([ // This is a comment
+    array(
+        'HelloHelloHelloHelloHelloHelloHelloHelloHelloHelloHelloHelloHelloHelloHelloHello',
+        11212,
+        112.1,
+        true,
+        $bar,
+        PHP_VERSION,
+    ),
+    array('World', 125.1, 12, false, $quux, PHP_VERSION_ID),
+    array('!!', 125.1, 123512, false, $bar, PHP_VERSION_ID),
+]);
+
+// too small for table.
+$a = [[1, 2], [3, 4], [5, 6]];
+
+$arr = [
+    ['الاسم',    'العمر', 'المدينة', 'المهنة'],
+    ['أحمد',    '30',    'الرياض',  'مهندس'],
+    ['فاطمة',   '25',    'جدة',     'طبيبة'],
+    ['علي',     '35',    'الدمام',  'محاسب'],
+    ['ليلى',    '28',    'مكة',     'معلمة'],
+    ['خالد',    '40',    'المدينة', 'مدير'],
+    ['سارة',    '22',    'تبوك',    'طالبة'],
+    ['يوسف',    '32',    'حائل',    'مبرمج'],
+    ['نورة',    '29',    'أبها',    'مصممة'],
+    ['عبدالله', '38',    'جازان',   'محامٍ'],
+];
+
+$arr = [
+    ['Name',    'Age', 'City',     'Occupation'],
+    ['John',    '30',  'New York', 'Engineer'],
+    ['Jane',    '25',  'London',   'Doctor'],
+    ['Mike',    '35',  'Paris',    'Accountant'],
+    ['Emily',   '28',  'Tokyo',    'Teacher'],
+    ['David',   '40',  'Sydney',   'Manager'],
+    ['Sarah',   '22',  'Toronto',  'Student'],
+    ['Robert',  '32',  'Berlin',   'Programmer'],
+    ['Jessica', '29',  'Rome',     'Designer'],
+    ['William', '38',  'Madrid',   'Lawyer'],
+];
+
+$arr = [
```

Comment:
top row, third column looks wrong?

Reply:
The alignment problem arises from mixing Chinese characters with numbers. They occupy different horizontal spaces, making perfect alignment not possible. If you replace the numbers in the second column with Chinese characters, you'll see it align correctly. 

This is a common issue when combining ideographic scripts like Chinese with Latin characters or numbers, due to their inherent width differences.